qt/widgets/common/src/FunctionModel.cpp

```diff
@@ -0,0 +1,404 @@
+}
+
+int MultiDomainFunctionModel::currentDomainIndex() const {
+  return static_cast<int>(m_currentDomainIndex);
+}
+
+void MultiDomainFunctionModel::setCurrentDomainIndex(int index) {
+  checkIndex(index);
+  m_currentDomainIndex = static_cast<size_t>(index);
+}
+
+double MultiDomainFunctionModel::getLocalParameterValue(const QString &parName,
+                                                        int i) const {
+  return getSingleFunction(i)->getParameter(parName.toStdString());
+}
+
+bool MultiDomainFunctionModel::isLocalParameterFixed(const QString &parName,
+                                                     int i) const {
+  auto fun = getSingleFunction(i);
+  auto const parIndex = fun->parameterIndex(parName.toStdString());
+  return fun->isFixed(parIndex);
+}
+
+QString MultiDomainFunctionModel::getLocalParameterTie(const QString &parName,
+                                                       int i) const {
+  auto fun = getSingleFunction(i);
+  auto const parIndex = fun->parameterIndex(parName.toStdString());
+  auto const tie = fun->getTie(parIndex);
+  if (!tie)
+    return "";
+  auto const tieStr = QString::fromStdString(tie->asString());
+  auto const j = tieStr.indexOf('=');
+  return tieStr.mid(j + 1);
+}
+
+void MultiDomainFunctionModel::setLocalParameterValue(const QString &parName,
+                                                      int i, double value) {
+  getSingleFunction(i)->setParameter(parName.toStdString(), value);
+}
+
+void MultiDomainFunctionModel::setLocalParameterValue(const QString &parName,
+                                                      int i, double value,
+                                                      double error) {
+  auto fun = getSingleFunction(i);
+  auto const parIndex = fun->parameterIndex(parName.toStdString());
+  fun->setParameter(parIndex, value);
+  fun->setError(parIndex, error);
+}
+
+void MultiDomainFunctionModel::setLocalParameterFixed(const QString &parName,
+                                                      int i, bool fixed) {
+  auto fun = getSingleFunction(i);
+  auto const parIndex = fun->parameterIndex(parName.toStdString());
+  if (fixed) {
+    fun->fix(parIndex);
+  } else if (fun->isFixed(parIndex)) {
+    fun->unfix(parIndex);
+  }
+}
+
+void MultiDomainFunctionModel::setLocalParameterTie(const QString &parName,
+                                                    int i, QString tie) {
+  auto fun = getSingleFunction(i);
+  auto const name = parName.toStdString();
+  if (tie.isEmpty()) {
+    fun->removeTie(fun->parameterIndex(name));
+  } else {
+    auto const j = tie.indexOf('=');
+    fun->tie(name, tie.mid(j + 1).toStdString());
+  }
+}
+
+void MultiDomainFunctionModel::changeTie(const QString &parName,
+                                         const QString &tie) {
+  try {
+    setLocalParameterTie(parName, static_cast<int>(m_currentDomainIndex), tie);
+  } catch (std::exception &) {
+    // the tie is probably being edited
```

Comment:
Does any action need to be taken here or are we ok just to catch the exception?

Reply:
The method is called every time the user presses a key. Most of the time the error is due to unfinished editing.